service: backend-proposal-v2
frameworkVersion: '3'
configValidationMode: error

provider:
  name: aws
  runtime: python3.11
  stage: dev
  region: us-east-1
  environment:
    COGNITO_USER_POOL_ID: !Ref CognitoUserPool
    COGNITO_USER_POOL_CLIENT_ID: !Ref CognitoPasswordlessUserPoolClient

  iam:
    role:
      statements:
        - Effect: Allow
          Action:
            - cognito-idp:*
          Resource: "*"

custom:
  userPoolName: ${self:provider.stage}-passwordless-user-pool

resources:
  Resources:
    CognitoUserPool: ${file(./src/resources/auth/serverless.yml):CognitoUserPool}
    CognitoPasswordlessUserPoolClient: ${file(./src/resources/auth/serverless.yml):CognitoPasswordlessUserPoolClient}

functions:

  defineAuthChallenge:
    handler: src/adapters/primary/auth/define_auth_challenge.handler
    events:
      - cognitoUserPool:
          pool: ${self:resources.Resources.CognitoUserPool.Properties.UserPoolName}
          trigger: DefineAuthChallenge
          existing: true

  create_organization:
    handler: src/adapters/primary/create_organization/index.handler
    events:
      - http:
          path: /organizations/create
          method: post
          cors: true
          private: true
          response:
            headers:
              Content-Type: "'application/json'"
    environment:
      ENV: ${self:provider.stage}

plugins:
  - serverless-offline

